manager/state/raft/raft.go

```diff
@@ -91,6 +91,10 @@ type Node struct {
 	removed     uint32
 	joinAddr    string
 
+	// proposeLock must be held when keeping track of proposals.
+	proposeLock sync.Mutex
+	proposals   map[uint64]context.CancelFunc
```

Comment:
Maybe we should set a flag in finalize to prevent new proposals from starting afterwards.